virtualizarr/vendor/kerchunk/utils.py

```diff
@@ -0,0 +1,81 @@
+import base64
+
+import fsspec
+import ujson
+import zarr
+
+
+def _encode_for_JSON(store):
+    """Make store JSON encodable"""
+    for k, v in store.copy().items():
+        if isinstance(v, list):
+            continue
+        else:
+            try:
+                # minify JSON
+                v = ujson.dumps(ujson.loads(v))
+            except (ValueError, TypeError):
+                pass
+            try:
+                store[k] = v.decode() if isinstance(v, bytes) else v
+            except UnicodeDecodeError:
+                store[k] = "base64:" + base64.b64encode(v).decode()
+    return store
+
+
+def _inline_array(group, threshold, names, prefix=""):
+    for name, thing in group.items():
+        if prefix:
+            prefix1 = f"{prefix}.{name}"
+        else:
+            prefix1 = name
+        if isinstance(thing, zarr.Group):
+            _inline_array(thing, threshold=threshold, prefix=prefix1, names=names)
+        else:
+            cond1 = threshold and thing.nbytes < threshold
+            cond2 = prefix1 in names
+            if cond1 or cond2:
+                original_attrs = dict(thing.attrs)
+                arr = group.create_dataset(
+                    name=name,
+                    dtype=thing.dtype,
+                    shape=thing.shape,
+                    data=thing[:],
+                    chunks=thing.shape,
+                    compression=None,
+                    overwrite=True,
+                    fill_value=thing.fill_value,
+                )
+                arr.attrs.update(original_attrs)
+
+
+def inline_array(store, threshold=1000, names=None, remote_options=None):
```

Comment:
We probably don't need this at all, because we never use kerchunk's inlining (we use `loadable_variables` instead).